src/helm/benchmark/adaptation/adapters/language_modeling_adapter.py

```diff
@@ -48,10 +48,20 @@ def generate_requests(self, eval_instance: Instance) -> List[RequestState]:
         """
         max_sequence_length: int = self.window_service.max_sequence_length
         max_request_length: int = self.window_service.max_request_length
-        # Handle max_sequence_and_tokens_length
+        max_sequence_and_generated_tokens_length: int = self.window_service.max_sequence_and_generated_tokens_length
+        max_tokens: int = (
+            min(self.adapter_spec.max_tokens, self.window_service.max_generated_max_tokens)
+            if self.adapter_spec.max_tokens > 0
+            else self.adapter_spec.max_tokens
+        )
+        # Here we assume that there is a max_request_and_generated_tokens_length that we can
+        # compute from max_sequence_length, max_request_length, and max_sequence_and_generated_tokens_length.
         max_request_length = min(
             max_request_length,
-            self.window_service.max_sequence_and_generated_tokens_length - self.adapter_spec.max_tokens,
+            max_request_length
+            - max_sequence_length
+            + max_sequence_and_generated_tokens_length
+            - self.adapter_spec.max_tokens,
```

Comment:
Add a comment explaining the reasoning behind this?

Should `self.adapter_spec.max_tokens` be `max_tokens`?

Can you just define `max_request_and_generated_tokens_length` and have this be `max_request_and_generated_tokens_length - self.adapter_spec.max_tokens` rather than adding a `max_request_length - max_sequence_length` term?

Do you also have to do a similar adjustment to `max_sequence_length`?

Are prefix and suffix tokens accounted for correctly; which lengths include them and which do not?